crates/meroctl/src/cli/context/update.rs

```diff
@@ -0,0 +1,282 @@
+    context_id: ContextId,
+
+    #[clap(
+        long,
+        short = 'a',
+        help = "The application ID to update in the context"
+    )]
+    application_id: Option<ApplicationId>,
+
+    #[clap(
+        long,
+        conflicts_with = "application_id",
+        help = "Path to the application file to watch and install locally"
+    )]
+    path: Option<Utf8PathBuf>,
+
+    #[clap(long, help = "Metadata needed for the application installation")]
+    metadata: Option<String>,
+
+    #[clap(
+        long,
+        short = 'w',
+        conflicts_with = "application_id",
+        requires = "path"
+    )]
+    watch: bool,
+
+    #[arg(long = "as", help = "Public key of the executor")]
+    pub executor: PublicKey,
+}
+
+impl UpdateCommand {
+    pub async fn run(self, environment: &Environment) -> EyreResult<()> {
+        let config = load_config(&environment.args.home, &environment.args.node_name)?;
+        let multiaddr = fetch_multiaddr(&config)?;
+        let client = Client::new();
+
+        match self {
+            Self {
+                context_id,
+                application_id: Some(application_id),
+                path: None,
+                metadata: None,
+                watch: false,
+                executor: executor_public_key,
+            } => {
+                update_context_application(
+                    environment,
+                    &client,
+                    multiaddr,
+                    context_id,
+                    application_id,
+                    &config.identity,
+                    executor_public_key,
+                )
+                .await?;
+            }
+            Self {
+                context_id,
+                application_id: None,
+                path: Some(path),
+                metadata: None,
+                watch: false,
+                executor: executor_public_key,
+            } => {
+                install_app_and_update_context(
+                    environment,
+                    &client,
+                    multiaddr,
+                    path,
+                    context_id,
+                    None,
+                    &config.identity,
+                    executor_public_key,
+                )
+                .await?;
+            }
+
+            Self {
+                context_id,
+                application_id: None,
+                path: Some(path),
+                metadata,
+                watch: true,
+                executor: executor_public_key,
+            } => {
+                watch_app_and_update_context(
+                    environment,
+                    &client,
+                    multiaddr,
+                    context_id,
+                    path,
+                    metadata.map(String::into_bytes),
+                    &config.identity,
+                    executor_public_key,
+                )
+                .await?;
+            }
```

Comment:
the only difference between these two would be when `watch` is true, call `watch_app*`

but they should both "install" first